### PR TITLE
added info, natrules, firewallrules on gateway

### PR DESCRIPTION
**Pull request by floriankammermann**
added the possibility to retrieve the natrules and firewallrules of an edge gateway. For completeness I added info on the gateway command.

---
This change is [Reviewable](https://reviewable.io/reviews/vmware/vcd-cli/190)

### CODE REVIEW

**Comment by tsugliani** [comment]
very useful ! 👍

File: vcd_cli/gateway.py [hunk 1]
```diff
@@ -48,3 +54,131 @@ def list_gateways(ctx):
...
+@click.argument('name',
+                metavar='<name>',
+                required=True)
+def info(ctx, name):
+    restore_session(ctx, vdc_required=True)
+    client = ctx.obj['client']
+    vdc_href = ctx.obj['profiles'].get('vdc_href')
+    vdc = VDC(client, href=vdc_href)
+    edge_gateways = vdc.list_edge_gateways()
+    for gw in edge_gateways:
```
Comment (sahithi) [inline vcd_cli/gateway.py]: rename gw ->edge (or) edge_gateway

File: vcd_cli/gateway.py [hunk 2]
```diff
@@ -48,3 +54,131 @@ def list_gateways(ctx):
...
+        if name == gw.get('name'):
+            resource = client.get_resource(gw.get('href'))
+            result = edge_gateway_to_dict(resource)
+            stdout(result, ctx)
+            return
+    raise Exception('not found')
+
+
+def edge_gateway_to_dict(gw):
```
Comment (sahithi) [inline vcd_cli/gateway.py]: rename gw -> edge or edge_gateway

File: vcd_cli/gateway.py [hunk 3]
```diff
@@ -48,3 +54,131 @@ def list_gateways(ctx):
...
+def info(ctx, name):
+    restore_session(ctx, vdc_required=True)
+    client = ctx.obj['client']
+    vdc_href = ctx.obj['profiles'].get('vdc_href')
+    vdc = VDC(client, href=vdc_href)
+    edge_gateways = vdc.list_edge_gateways()
+    for gw in edge_gateways:
+        if name == gw.get('name'):
+            resource = client.get_resource(gw.get('href'))
+            result = edge_gateway_to_dict(resource)
```
Comment (sahithi) [inline vcd_cli/gateway.py]: edge_gateway_to_dict(..) basically retrieves only name and href. Info command should display much more than that.
Any specific reason why you had to use new method edge_gateway_to_dict()? 
Have you tried vcd_cli.utils.to_dict()?

fyi, these utility methods should go to utils.py file

File: vcd_cli/gateway.py [hunk 4]
```diff
@@ -48,3 +54,131 @@ def list_gateways(ctx):
...
+                required=True)
+def info(ctx, name):
+    restore_session(ctx, vdc_required=True)
+    client = ctx.obj['client']
+    vdc_href = ctx.obj['profiles'].get('vdc_href')
+    vdc = VDC(client, href=vdc_href)
+    edge_gateways = vdc.list_edge_gateways()
+    for gw in edge_gateways:
+        if name == gw.get('name'):
+            resource = client.get_resource(gw.get('href'))
```
Comment (sahithi) [inline vcd_cli/gateway.py]: resource -> edge_resource

File: vcd_cli/gateway.py [hunk 5]
```diff
@@ -48,3 +54,131 @@ def list_gateways(ctx):
...
+            result = edge_gateway_nat_rules_to_dict(resource)
+            stdout(result, ctx)
+            return
```
Comment (sahithi) [inline vcd_cli/gateway.py]: Move this logic to a new method in "vdc.get_nat_rules(edge_gateway_name)". It should ideally go to edge_gateway but as there is no existing class, you can move it to vdc.py.

File: vcd_cli/gateway.py [hunk 6]
```diff
@@ -48,3 +54,131 @@ def list_gateways(ctx):
...
+    raise Exception('not found')
+
+
+def edge_gateway_nat_rules_to_dict(gw):
+    result = []
+    for nr in gw.Configuration.EdgeGatewayServiceConfiguration.NatService.NatRule:
```
Comment (sahithi) [inline vcd_cli/gateway.py]: Missing Attribute or None checks. Refer utils.pvdc_to_dict()  and other to_dict() methods to get an idea.
Variables have to be more descriptive ex: nr, gnr, nrd. Please do this across the change-set.

File: vcd_cli/gateway.py [hunk 7]
```diff
@@ -48,3 +54,131 @@ def list_gateways(ctx):
...
+    client = ctx.obj['client']
+    vdc_href = ctx.obj['profiles'].get('vdc_href')
+    vdc = VDC(client, href=vdc_href)
+    edge_gateways = vdc.list_edge_gateways()
+    for gw in edge_gateways:
+        if name == gw.get('name'):
+            resource = client.get_resource(gw.get('href'))
+            result = edge_gateway_firewall_rules_to_dict(resource)
+            stdout(result, ctx)
+            return
```
Comment (sahithi) [inline vcd_cli/gateway.py]: Move the logic to new method vdc.get_firewall_rules(edge_gateway_name)

File: vcd_cli/gateway.py [hunk 8]
```diff
@@ -29,7 +29,13 @@ def gateway(ctx):
 \b
     Examples
         vcd gateway list
-            Get list of edge gateways in current virtual datacenter.
+            Get list of edge gateways in current the virtual datacenter.
+\b
```
Comment (sahithi) [inline vcd_cli/gateway.py]: Missing example for info command

**Pull request closed without merging**